packages/react-hooks/README.md

```diff
@@ -1,11 +1,301 @@
+[
+	(...args: A) => void,
+	{
+		data: R | null,
+		error: Error | null,
+		loading: boolean,
+		calledOnce: boolean
+	}
+]
 ```
-const reactHooks = require('react-hooks');
+#### Example
+```javascript
+import { useAsync } from "@w2solutions/react-hooks"
 
-// TODO: DEMONSTRATE API
+const AsyncInvocation = () => {
+	const [counter, setCounter] = useState(0)
+
+	const asyncFunction = (...args: any[]): Promise<any> => {
+		return new Promise((resolve) => {
+			setTimeout(() => {
+				resolve(args)
+			}, 1000)
+		})
+	}
+
+	const callbackFn = useCallback(() => {
+		return asyncFunction(counter)
+	}, [counter])
+
+	const [callAsyncFunction, stateData] = useAsync(callbackFn)
+
+	return (
+		<>
+			<button onClick={() => setCounter(counter+1)}>
+				Counter +1
+			</button>
+			<button disabled={stateData.calledOnce} onClick={() => callAsyncFunction()}>
+				Call asynchronous function
+			</button>
+			<div style={stateData.loading ? {display: "block"} : {display: "none"}}>
+				Loading...
+			</div>
+			<div>Data: {stateData.data}</div>
+			<div>Error: {stateData.error}</div>
+		</>
+	)
+}
+```
+
+> `asyncFunction` resolves after 1 second. The `asyncFunction` is called within `useCallback` hook, which is passed to the `useAsync` hook. The `calledOnce` flag disables the button, so that the asynchronous function can only be invoked once by this button. `Loading` flag makes a loading div visible.
+
+
+## `useDebounceEffect(callback, delay, dependencyArray)`
+`useDebounceEffect` invokes a callback function after a given delay if any item in the dependency array has been changed.
+#### Input
+```javascript
+callback: EffectCallback
+delay: number
+dependencyArray?: DependencyList
 ```
+#### Example
+```javascript
+import { useDebounceEffect } from "@w2solutions/react-hooks"
+
+const DebounceEffect = () => {
+	const [counter, setCounter] = useState(1)
+	const [anotherCounter, setAnotherCounter] = useState(1)
+
+	const square = useCallback(()=>{
+		setAnotherCounter(counter**2)
+	},[counter])
+
+	useDebounceEffect(square, 300, [counter])
+
+	return (
+		<>
+			<button onClick={() => setCounter(counter + 1)}>
+				Counter +1
+			</button>
+			<div>{counter}</div>
+			<div>{anotherCounter}</div>
+		</>
+	)
+}
+```
+>300 ms after the button is clicked, the callback function `square` is invoked, which squares the value of counter and assigns the result to the state of `anotherCounter`.
+## `useFileSelect(options)`
+`useFileSelect` selects the file(s) and returns a promise, which resolves with a `FileList`.
+#### Input
+```javascript
+options: {
+	multiple: boolean,
+	accept: string
+}
+```
+#### Output
+```javascript
+() => Promise<FileList>
+```
+#### Example
+```javascript
+import { useFileSelect } from "@w2solutions/react-hooks"
+
+const FileSelector = () => {
+	const  selectFiles = useFileSelect(
+		{multiple: false, accept: ".txt"}
+	)
+
+	const [fileName, setFileName] = useState<string | null>(null)
+
+	const callSelectFiles = useCallback(async () => {
```

Comment:
rename to `handleClick` and remove `useCallback`. 
`useCallback` is not required here.